### PR TITLE
Include the legacy namespace identifier in both interface prototype objects and platform objects.

### PR DESCRIPTION
**Pull request by Ms2ger**
***
<a href="https://pr-preview.s3.amazonaws.com/Ms2ger/webidl/pull/577.html" title="Last updated on Sep 20, 2018, 8:59 AM GMT (4ad138f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/heycam/webidl/577/4afb5e8...Ms2ger:4ad138f.html" title="Last updated on Sep 20, 2018, 8:59 AM GMT (4ad138f)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1076,6 +1076,19 @@ with a [{{NoInterfaceObject}}] [=extended attribute=],
...
 
+<div algorithm>
+
+The <dfn>qualified name</dfn> of an [=interface=] |I| is defined as follows:
+
+1.  Let |interface| be the [=identifier=] of |I|.
+1.  If |I| has a [{{LegacyNamespace}}] [=extended attribute=], then:
+    1.  Let |namespace| be the identifier argument of the [{{LegacyNamespace}}]
+        [=extended attribute=].
+    1.  Return the concatenation of |namespace|, ".", and |interface|.
```
Comment (tobie) [inline index.bs]: Maybe link [infra concatenate](https://infra.spec.whatwg.org/#string-concatenate)?

Comment (Ms2ger) [reply index.bs]: I always forget that exists. Thanks.

**Comment by domenic** [comment]
The title of this PR seems to be more like #357, but the diff seems to be about namespace objects only.

File: index.bs [hunk 2]
```diff
@@ -1076,6 +1076,20 @@ with a [{{NoInterfaceObject}}] [=extended attribute=],
 and [=callback interfaces=] which declare [=constants=]
 must be annotated with an [{{Exposed}}] [=extended attribute=].
 
+<div algorithm>
+
+The <dfn>qualified name</dfn> of an [=interface=] |I| is defined as follows:
+
+1.  Let |interface| be the [=identifier=] of |I|.
```
Comment (TimothyGu) [inline index.bs]: I'd encourage _not_ using any one-letter variables any more. How about "Let *identifier* be the identifier of *interface*"?

File: index.bs [hunk 3]
```diff
@@ -1076,6 +1076,20 @@ with a [{{NoInterfaceObject}}] [=extended attribute=],
...
+1.  If |I| has a [{{LegacyNamespace}}] [=extended attribute=], then:
+    1.  Let |namespace| be the identifier argument of the [{{LegacyNamespace}}]
+        [=extended attribute=].
+    1.  Return the [=concatenation=] of « |namespace|, |interface| » with
+        separator ".".
```
Comment (TimothyGu) [inline index.bs]: "`.`"

Comment (TimothyGu) [reply index.bs]: Or, even better, U+002E (.)

**Comment by Ms2ger** [comment]
Review ping?

File: index.bs [hunk 4]
```diff
@@ -1076,6 +1076,20 @@ with a [{{NoInterfaceObject}}] [=extended attribute=],
...
+<div algorithm>
+
+The <dfn>qualified name</dfn> of an [=interface=] |interface| is defined as follows:
+
+1.  Let |identifier| be the [=identifier=] of |interface|.
+1.  If |interface| has a [{{LegacyNamespace}}] [=extended attribute=], then:
+    1.  Let |namespace| be the identifier argument of the [{{LegacyNamespace}}]
+        [=extended attribute=].
+    1.  Return the [=concatenation=] of « |namespace|, |identifier| » with
+        separator <span class="char">U+002E FULL STOP (".")</span>.
```
Comment (TimothyGu) [inline index.bs]: A bit unsure if (".") needs to be in the `<span class=char>` as well…

Comment (Ms2ger) [reply index.bs]: Looking at the existing spec for about ten seconds yields
```html
<span class="char">U+005F LOW LINE ("_")</span>
```

**Comment by annevk** [comment]
I'd be happy to rebase & merge this provided we don't enforce commit message guidelines for IDL. @domenic?

**Pull request merged**